### PR TITLE
CPFP fixups and followups

### PR DESCRIPTION
**Pull request by darosior**
This is a followup to #302, fixing inconsistencies and cleaning up a bit.
The revault_tx counterpart is at https://github.com/revault/revault_tx/pull/117 .

### CODE REVIEW

File: tests/test_misc.py [hunk 1]
```diff
@@ -229,41 +229,59 @@ def test_no_cosig_server(revault_network):
...
+
 @pytest.mark.skipif(not POSTGRES_IS_SETUP, reason="Needs Postgres for servers db")
 def test_cpfp_transaction(revault_network, bitcoind):
     CSV = 12
     revault_network.deploy(
         2,
         1,
         csv=CSV,
-        with_watchtowers=False,
+        # with_watchtowers=False,
```
Comment (danielabrozzoni) [inline tests/test_misc.py:253]: Don't comment, either you set to true or to false

Comment (danielabrozzoni) [reply tests/test_misc.py:253]: I don't recall any real reason, probably I followed a "if watchtowers are not needed here why spawning them at all" kinda reasoning to make the test run quicker

File: src/daemon/bitcoind/poller.rs [hunk 2]
```diff
@@ -392,30 +399,75 @@ fn mark_confirmed_emers(
...
+
+// CPFP a bunch of transactions, bumping their feerate by at least `target_feerate`.
+// `target_feerate` is expressed in sat/kWU.
+// All the transactions' feerate MUST be below `target_feerate`.
 fn cpfp_package(
     revaultd: &Arc<RwLock<RevaultD>>,
     bitcoind: &BitcoinD,
-    mut tx_package: Vec<impl CpfpableTransaction>,
-    current_feerate: u64,
+    mut tx_package: Vec<ToBeCpfped>,
```
Comment (danielabrozzoni) [inline src/daemon/bitcoind/poller.rs:444]: Can we call this a package?

File: tests/test_misc.py (hunk 1 above)
Comment (darosior) [reply tests/test_misc.py:253]: Makes sense. My own rationale for making it default to `True` was: "test in as many scenarii as possible".

File: src/daemon/bitcoind/poller.rs (hunk 2 above)
Comment (darosior) [reply src/daemon/bitcoind/poller.rs:444]: Not really, i fix it elsewhere IIRC. If not here, the rationale was that i didn't want to increase the diff by too much.

**Comment by darosior** [comment]
Fixed your comment on the tests.

**Comment by darosior** [comment]
Rebased on master for Github aesthetics

**Review by danielabrozzoni — COMMENTED** [review]
ACK 5ae472c2c995cd3297e1249e07e287fe3e9ddf3e

I'll re-ack if you decide to set the revault_tx reference to master :)

**Comment by danielabrozzoni** [comment]
Uhm I just saw https://github.com/revault/revaultd/pull/302#discussion_r767336107, I'll fix in #331 after this is merged

**Comment by darosior** [comment]
Yeah i need to update revault_net first

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by darosior** [comment]
Update revault_tx and revault_net to their respective master

**Comment by darosior** [comment]
Spurious HTTP timeout when generating blocks on CI: added a commit (unrelated to this PR, but hey CI won't be flaky then) making the timeout much larger so we are now fine with generating loads of blocks at once.

**Pull request merged**